pkg/collector/corechecks/network-devices/cisco-sdwan/sdwan.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		log.Warnf("Error getting devices from Cisco SD-WAN API: %s", err)
+	}
+	vEdgeInterfaces, err := c.client.GetVEdgeInterfaces()
+	if err != nil {
+		log.Warnf("Error getting vEdge interfaces from Cisco SD-WAN API: %s", err)
+	}
+	cEdgeInterfaces, err := c.client.GetCEdgeInterfaces()
+	if err != nil {
+		log.Warnf("Error getting cEdge interfaces from Cisco SD-WAN API: %s", err)
+	}
+	deviceStats, err := c.client.GetDeviceHardwareMetrics()
+	if err != nil {
+		log.Warnf("Error getting device metrics from Cisco SD-WAN API: %s", err)
+	}
+	interfaceStats, err := c.client.GetInterfacesMetrics()
+	if err != nil {
+		log.Warnf("Error getting interface metrics from Cisco SD-WAN API: %s", err)
+	}
+	appRouteStats, err := c.client.GetApplicationAwareRoutingMetrics()
+	if err != nil {
+		log.Warnf("Error getting application-aware routing metrics from Cisco SD-WAN API: %s", err)
+	}
+	controlConnectionsState, err := c.client.GetControlConnectionsState()
+	if err != nil {
+		log.Warnf("Error getting control-connection states from Cisco SD-WAN API: %s", err)
+	}
+	ompPeersState, err := c.client.GetOMPPeersState()
+	if err != nil {
+		log.Warnf("Error getting OMP peer states from Cisco SD-WAN API: %s", err)
+	}
+	bfdSessionsState, err := c.client.GetBFDSessionsState()
+	if err != nil {
+		log.Warnf("Error getting BFD session states from Cisco SD-WAN API: %s", err)
+	}
+	deviceCounters, err := c.client.GetDevicesCounters()
+	if err != nil {
+		log.Warnf("Error getting device counters from Cisco SD-WAN API: %s", err)
+	}
+
+	devicesMetadata, deviceTags, uptimes := payload.ProcessDevices(c.config.Namespace, devices)
+	interfacesMetadata, ipAddressesMetadata, interfacesMap := payload.ProcessInterfaces(c.config.Namespace, vEdgeInterfaces, cEdgeInterfaces)
+
+	c.metricsSender.SetDeviceTags(deviceTags)
+
+	if !c.config.NoMetadata {
+		c.metricsSender.SendMetadata(devicesMetadata, interfacesMetadata, ipAddressesMetadata)
+	}
+	c.metricsSender.SendDeviceMetrics(deviceStats)
+	c.metricsSender.SendInterfaceMetrics(interfaceStats, interfacesMap)
+	c.metricsSender.SendUptimeMetrics(uptimes)
+	c.metricsSender.SendAppRouteMetrics(appRouteStats)
+	c.metricsSender.SendControlConnectionMetrics(controlConnectionsState)
+	c.metricsSender.SendOMPPeerMetrics(ompPeersState)
+	c.metricsSender.SendBFDSessionMetrics(bfdSessionsState)
+	c.metricsSender.SendDeviceCountersMetrics(deviceCounters)
+
+	// Commit
+	c.metricsSender.Commit()
+	log.Info("Done running Cisco SD-WAN check")
+
+	return nil
+}
+
+// Configure the Cisco SD-WAN check
+func (c *CiscoSdwanCheck) Configure(senderManager sender.SenderManager, integrationConfigDigest uint64, rawInstance integration.Data, rawInitConfig integration.Data, source string) error {
+	// Must be called before c.CommonConfigure
+	c.BuildID(integrationConfigDigest, rawInstance, rawInitConfig)
+
+	err := c.CommonConfigure(senderManager, integrationConfigDigest, rawInitConfig, rawInstance, source)
```

Comment:
CommonConfigure will set the collection interval to 15s by default. 
Any idea how we could override this with a higher value? A 15s collection interval will probably be useless for this integration

Reply:
You can look at how network path does it here: https://datadoghq.atlassian.net/browse/NDMII-2570?atlOrigin=eyJpIjoiN2FmYmJlOGQxN2U0NDQwYmI1NjVlZDg1ZGVlNmJmYjYiLCJwIjoiaiJ9

Reply:
Perfect thanks!